daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -189,6 +189,11 @@ object Speedy {
       // Restore the frame and actuals to there state when the continuation was created.
       frame = frameToBeRestored
       actuals = actualsToBeRestored
+      // And restore the env-stack.
+      restoreEnvStack(envSizeToBeRestored)
+    }
+
+    @inline def restoreEnvStack(envSizeToBeRestored: Int): Unit = {
```

Comment:
If we follow my suggestions below, we don't need this split anymore.

Reply:
Indeed. Thank you.